bin/test

```diff
@@ -37,40 +37,66 @@ def putv(message)
   end
 end
 
-failed_libs = []
+def putci(message)
+  puts message if ENV["CI"]
+end
+
+passed_gems = []
+failed_gems = []
 
 dirs.each do |dir|
-  test_path = dir + "_scripts/test"
+  test_script = dir + "_scripts/test"
   case
-  when !test_path.file?
-    puts "Skipping: cannot find test script at `#{test_path}`..."
-  when !test_path.executable?
-    puts "Skipping: `#{test_path}` is not executable..."
+  when !dir.exist?
+    puts "Skipping: `#{dir}` is not found..."
+    putci "::warning ::`#{dir}` is not found"
+  when !dir.directory?
+    puts "Skipping: `#{dir}` is not a directory..."
+    putci "::warning ::`#{dir}` is not a directory"
+  when !test_script.exist?
+    puts "Skipping: `#{test_script}` is not found..."
+    putci "::warning ::`#{test_script}` is not found"
+  when !test_script.file?
+    puts "Skipping: `#{test_script}` is not a file..."
+    putci "::warning ::`#{test_script}` is not a file"
+  when !test_script.executable?
+    puts "Skipping: `#{test_script}` is not executable..."
+    putci "::warning ::`#{test_script}` is not executable"
   else
-    puts "::group::Test for #{dir}" if ENV['CI']
-    puts "Detected test script: #{test_path}"
+    putci "::group::Test for #{dir}"
+    puts "Detected test script: #{test_script}"
 
     env = {
       "REPO_ROOT" => @repo_root.to_s,
       "REPOS" => @repos.join(File::PATH_SEPARATOR),
       "RBS_DIR" => dir.to_s
     }
-    stdout, status = Open3.capture2(env, test_path.to_s)
-
-    putv(stdout)
+    status = Open3.popen2e(env, test_script.to_s) do |_stdin, stdout_stderr, thread|
+      putv stdout_stderr
+      thread.value
+    end
```

Comment:
Just confirmation, the reasons for replacing `capture2` with `popen2e` are the following two things, right?


* To print stderr by changing `2` to `2e` method.
* To print output before the command finishes.

I agree with this change but I'd like to confirm the intention.

Reply:
Yes, that's right! I'm very sorry for missing the change reason. 🙇🏼